lib/retriable.rb

```diff
@@ -6,14 +6,22 @@
 module Retriable
   module_function
 
-  def self.configure
+  def configure
     yield(config)
   end
 
   def config
     @config ||= Config.new
   end
 
+  def retriable_with_context(context_key, options = {}, &block)
+    if !config.context.key?(context_key)
+      raise ArgumentError, "#{context_key} not found in Retriable.config.context"
+    end
+
+    retriable(config.context[context_key].merge(options), &block) if block
+  end
+
   def retriable(opts = {})
```

Comment:
yeah you don't need it, but it does make it clear that the method takes a block - just makes the method signature fully explicit because otherwise the `yield` is kinda buried way in there (where it needs to be, but it's a little hard to spot at first).  up to you really, though.